File: 0646-maximum-length-of-pair-chain/0646-maximum-length-of-pair-chain.cpp
class Solution {
public:
    static bool comp(vector<int> &a, vector<int> &b){
        return a[1] < b[1];
    }
    int findLongestChain(vector<vector<int>>& pairs) {
        sort(pairs.begin(), pairs.end(), comp);
        int cnt = 0, curr=INT_MIN;
        for(auto x : pairs){
            if(x[0] > curr){
                cnt++;
                curr = x[1];
            }
        }
        return cnt;
    }
};

File: 0646-maximum-length-of-pair-chain/0646-maximum-length-of-pair-chain.py
class Solution:
    def findLongestChain(self, pairs: List[List[int]]) -> int:
        pairs.sort(key = lambda x : x[1])
        cnt, curr = 0, float('-inf')
        for x in pairs:
            if x[0] > curr:
                cnt += 1
                curr = x[1]
        return cnt